camel/utils/commons.py

```diff
@@ -226,3 +238,131 @@ def to_pascal(snake: str) -> str:
 
 
 PYDANTIC_V2 = pydantic.VERSION.startswith("2.")
+
+
+def role_playing_with_function(
+    task_prompt: str = ("Assume now is 2024 in the Gregorian calendar, "
+                        "estimate the current age of University of Oxford "
+                        "and then add 10 more years to this age, "
+                        "and get the current weather of the city where "
+                        "the University is located."),
+    function_list: Optional[List] = None,
+    model_type=None,
+    chat_turn_limit=10,
+    assistant_role_name: str = "Searcher",
+    user_role_name: str = "Professor",
+) -> None:
+    r"""Initializes and conducts a `RolePlaying` with `FunctionCallingConfig`
+    session. The function creates an interactive and dynamic role-play session
+    where the AI Assistant and User engage based on the given task, roles, and
+    available functions. It demonstrates the versatility of AI in handling
+    diverse tasks and user interactions within a structured `RolePlaying`
+    framework.
+
+    Args:
+        task_prompt (str): The initial task or scenario description to start
+            the `RolePlaying` session. Defaults to a prompt involving the
+            estimation of KAUST's age and weather information.
+        function_list (list): A list of functions that the agent can utilize
+            during the session. Defaults to a combination of math, search, and
+            weather functions.
+        model_type (ModelType): The type of chatbot model used for both the
+            assistant and the user. Defaults to `GPT-4 Turbo`.
+        chat_turn_limit (int): The maximum number of turns (exchanges) in the
+            chat session. Defaults to 10.
+        assistant_role_name (str): The role name assigned to the AI Assistant.
+            Defaults to 'Searcher'.
+        user_role_name (str): The role name assigned to the User. Defaults to
+            'Professor'.
+
+    Returns:
+        None: This function does not return any value but prints out the
+            session's dialogues and outputs.
+    """
```

Comment:
Why the `role_playing_with_function`  example is moved to `utils`?

Reply:
Let's not put hard coded example into `utils`.

Reply:
> Why the `role_playing_with_function` example is moved to `utils`?

Hey @lightaime , in the RAG cookbook I used `role_playing_with_function` as example,


````
from camel.utils import role_playing_with_function
from camel.functions import MATH_FUNCS  # import another function from camel

role_playing_with_function(
    task_prompt=(
        "What is the main termination reasons for AI Society dataset, how many number of messages did camel decided to limit, what's the value plus 100?"
    ),
    function_list=[*RETRIEVER_FUNCS, *MATH_FUNCS],
)
````

 I think this function will be regularly used as a common example function to show how to implement tasks by using different functions with multi-agents, so I moved this function into utils, even this function has set default values, it still allows the user to input their own parameters including `task_prompt`, `function_list`,`model_type`, `chat_turn_limit`, `assistant_role_name`,`user_role_name` to adjust the behavior of this function with flexibility.